### PR TITLE
 Audit pagination api changes

### PR DESCRIPTION
**Pull request by ramyarv**
* Updating transaction json contracts
* Added pagination, chronological ordering, filters in audit api
* Added audit json contracts

### CODE REVIEW

File: functions/api/transactions.js [hunk 1]
```diff
@@ -181,4 +182,6 @@ async function hasNextPage (transactionCollection, snapshot){
     });
 }
 
-module.exports = router;
\ No newline at end of file
+module.exports = router;
+module.exports.hasPreviousPage = hasPreviousPage;
```
Comment (neerk28) [inline functions/api/transactions.js]: suggestion: its common to both transaction and audit. The funcs should be in a common place and not in one of the files.

**Review by neerk28 — APPROVED** [review]
LGTM

File: functions/api/audit.js [hunk 2]
```diff
@@ -16,18 +17,88 @@ router.get("/", isAdminOrSuperAdmin, async (request, response, next) => {
...
+        auditCollection = auditCollection
+                .limit(constants.PAGE_SIZE)
+    }
+    //Case 2 : only nextPageToken is given, then start after the doc id + pagesize
+    if(nextPageToken !== null ) {
+        var lastVisibleDoc = await db
+            .collection(constants.AUDIT)
+            .doc(nextPageToken)
+            .get()
+        console.log (lastVisibleDoc.id);
```
Comment (shrikantdange1208) [inline functions/api/audit.js:53]: Can add a log statement here

File: functions/api/transactions.js (hunk 1 above)
Comment (shrikantdange1208) [reply functions/api/transactions.js]: You can move it to utils.js. I have also added getNextDate and getPrevDate functions there.

**Review by shrikantdange1208 — DISMISSED** [review]
Couple of minor changes needed.

**Review by shrikantdange1208 — APPROVED** [review]
Approved

**Pull request merged**